dipy/viz/actor.py

```diff
@@ -0,0 +1,648 @@
+        vol = np.reshape(vol, [np.prod(vol.shape[:3]), vol.shape[3]])
+
+    uchar_array = numpy_support.numpy_to_vtk(vol, deep=0)
+    im.GetPointData().SetScalars(uchar_array)
+
+    if affine is None:
+        affine = np.eye(4)
+
+    # Set the transform (identity if none given)
+    transform = vtk.vtkTransform()
+    transform_matrix = vtk.vtkMatrix4x4()
+    transform_matrix.DeepCopy((
+        affine[0][0], affine[0][1], affine[0][2], affine[0][3],
+        affine[1][0], affine[1][1], affine[1][2], affine[1][3],
+        affine[2][0], affine[2][1], affine[2][2], affine[2][3],
+        affine[3][0], affine[3][1], affine[3][2], affine[3][3]))
+    transform.SetMatrix(transform_matrix)
+    transform.Inverse()
+
+    # Set the reslicing
+    image_resliced = vtk.vtkImageReslice()
+    set_input(image_resliced, im)
+    image_resliced.SetResliceTransform(transform)
+    image_resliced.AutoCropOutputOn()
+    image_resliced.SetInterpolationModeToLinear()
+    image_resliced.Update()
+
+    if nb_components == 1:
+        if lookup_colormap is None:
+            # Create a black/white lookup table.
+            lut = colormap_lookup_table((0, 255), (0, 0), (0, 0), (0, 1))
+        else:
+            lut = lookup_colormap
+
+    x1, x2, y1, y2, z1, z2 = im.GetExtent()
+    ex1, ex2, ey1, ey2, ez1, ez2 = image_resliced.GetOutput().GetExtent()
+
+    class ImageActor(vtk.vtkImageActor):
+
+        def input_connection(self, output):
+            if vtk.VTK_MAJOR_VERSION <= 5:
+                self.SetInput(output.GetOutput())
+            else:
+                self.GetMapper().SetInputConnection(output.GetOutputPort())
+            self.output = output
+            self.shape = (ex2 + 1, ey2 + 1, ez2 + 1)
+
+        def display_extent(self, x1, x2, y1, y2, z1, z2):
+            self.SetDisplayExtent(x1, x2, y1, y2, z1, z2)
+            if vtk.VTK_MAJOR_VERSION > 5:
+                self.Update()
+
+        def display(self, x=None, y=None, z=None):
+            if x is None and y is None and z is None:
+                self.display_extent(ex1, ex2, ey1, ey2, ez2/2, ez2/2)
+            if x is not None:
+                self.display_extent(x, x, ey1, ey2, ez1, ez2)
+            if y is not None:
+                self.display_extent(ex1, ex2, y, y, ez1, ez2)
+            if z is not None:
+                self.display_extent(ex1, ex2, ey1, ey2, z, z)
+
+        def opacity(self, value):
+            if vtk.VTK_MAJOR_VERSION <= 5:
+                self.SetOpacity(value)
+            else:
+                self.GetProperty().SetOpacity(value)
+
+        def copy(self):
+            im_actor = ImageActor()
+            im_actor.input_connection(self.output)
+            im_actor.SetDisplayExtent(*self.GetDisplayExtent())
+            im_actor.opacity(opacity)
+            return im_actor
+
+        def set_position(self, wx, wy, wz):
+            self.SetPosition(wx, wy, wz)
+
+        def get_position(self):
+            return self.GetPosition()
```

Comment:
Good point. I was thinking to add some simple functions like this to all our actors. But maybe this should happen at a different stage. And change it everywhere.